### PR TITLE
Fixes #13983: Remove template in system techniques

### PR DESCRIPTION
**Pull request by peckpeck**
https://issues.rudder.io/issues/13983

### CODE REVIEW

**Comment by peckpeck** [comment]
Commit modified

File: techniques/system/common/1.0/common.st [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    "directiveId"        string => "&TRACKINGKEY&";
+    "sharedfiles_folder" string => "&SHARED_FILES_FOLDER&";
+    "community_port"     string => "&COMMUNITYPORT&";
+    "uuid"               string => "&UUID&";
+    "davuser"                     string => "&DAVUSER&";
+    "davpw"                       string => "&DAVPASSWORD&";
+    "rudder_configuration_repository"
+                                  string => "&CONFIGURATION_REPOSITORY_FOLDER&";
+    "heartbeat_interval"    string => "&RUDDER_HEARTBEAT_INTERVAL&";
+    "rudder_node_config_id" string => "&RUDDER_NODE_CONFIG_ID&";
```
Comment (ncharles) [inline techniques/system/common/1.0/common.st]: could you align them all ? Or at least add a bit of consistency (davuser and davpw could be aligned with uuid)

File: techniques/system/common/1.0/metadata.xml [hunk 2]
```diff
@@ -19,13 +19,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
   <SYSTEM>true</SYSTEM>
 
   <TMLS>
-    <TML name="monitoring" />
     <TML name="cf-serverd"/>
```
Comment (ncharles) [inline techniques/system/common/1.0/metadata.xml]: you need to add common here

File: techniques/system/common/1.0/update.cf [hunk 3]
```diff
@@ -38,7 +38,8 @@ body copy_from remote(server, path)
         purge      => "true";
         copy_backup => "false";
     community_edition::
-        portnumber => "&COMMUNITYPORT&";
+# TODO why isn't this a constant ?
```
Comment (ncharles) [inline techniques/system/common/1.0/update.cf]: can you remove this todo ?

File: techniques/system/distributePolicy/1.0/common.st [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# Common variable for distribute policy system technique
+bundle common distribute_policy_common {
+  vars:
+    "directiveId"    string => "${distribute_policy_common.directiveId}";
+    "report_db_name" string => "&RUDDER_REPORTS_DB_NAME&";
+    "report_db_user" string => "&RUDDER_REPORTS_DB_USER&";
+    "relay_sync_method"      string => "&RELAY_SYNC_METHOD&";
+    "relay_sync_promises"    string => "&RELAY_SYNC_PROMISES&";
+    "relay_sync_sharedfiles" string => "&RELAY_SYNC_SHAREDFILES&";
+    "cmdb_endpoint" string => "&CMDBENDPOINT&";
```
Comment (ncharles) [inline techniques/system/distributePolicy/1.0/common.st]: can you also align these ?

File: techniques/system/server-roles/1.0/alive-check.cf [hunk 5]
```diff
@@ -59,18 +59,18 @@ bundle agent generic_alive_check(app)
 
     pass3.root_server::
 
-      "any" usebundle => rudder_common_report("server-roles", "result_success", "&TRACKINGKEY&", "Check ${app} status", "None", "The ${site_to_check} web interface is running"),
+      "any" usebundle => rudder_common_report("server-roles", "result_success", "${roles_common.directiveId}", "Check ${app} status", "None", "The ${site_to_check} web interface is running"),
```
Comment (ncharles) [inline techniques/system/server-roles/1.0/alive-check.cf]: it should be rudder_roles_common

File: techniques/system/server-roles/1.0/component-check.cf [hunk 6]
```diff
@@ -71,13 +71,13 @@ bundle agent root_component_check
       "any" usebundle => root_password_restart_jetty;
 
     !role_rudder_server_root::
-      "any" usebundle => rudder_common_report("${technique_name}", "result_na", "&TRACKINGKEY&",
+      "any" usebundle => rudder_common_report("${technique_name}", "result_na", "${roles_common.directiveId}",
```
Comment (ncharles) [inline techniques/system/server-roles/1.0/component-check.cf]: it should be rudder_roles_common

File: techniques/system/server-roles/1.0/common.st [hunk 7]
```diff
@@ -0,0 +1,6 @@
+# Common variable for server-role system technique
+bundle common rudder_roles_common {
```
Comment (ncharles) [inline techniques/system/server-roles/1.0/common.st]: you used roles_common everywhere, can you pick either one ?

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
Commit modified

**Comment by ncharles** [comment]
This is awesome !!! Thank you

**Comment by peckpeck** [comment]
OK, squash merging this PR

**Comment by peckpeck** [comment]
OK, merging this PR

**Pull request merged**